### PR TITLE
ci: remove mingw job

### LINKED ISSUE #18551: Windows CI: do we need msys2 / mingw ?
## Problem

Unnecessary CI builds increase the change of spurious failures, which are costly noise. Of course, we should fix all legitimate bugs, but we also cannot micro-manage every platform, so there needs to be a clear motivation for the CI builds that we maintain.

### Con

Reasons against maintaining a mingw CI job:

1. the windows mingw build is slow
2. has failures:
    - https://github.com/neovim/neovim/issues/18494
    - https://github.com/neovim/neovim/issues/18495
3. the mingw artifact is 10x bigger than the windows MSVC artifact: https://github.com/neovim/neovim/issues/10560
4. our releases have long used the MSVC (not mingw) artifact as the downloadable artifact for Windows users: https://github.com/neovim/neovim/releases
5. non-MSVCRT has limitations documented by libuv: http://docs.libuv.org/en/v1.x/process.html
   > On Windows file descriptors greater than 2 are available to the child process only if the child processes uses the MSVCRT runtime.

### Pro

Reasons in favor of continuing to maintain mingw CI job:

- the mingw build is/was how we run `gcov` in Windows CI, for code-coverage
- extra CI coverage for weird corner cases
    - counterpoint: very, very rare that mingw test failure leads to a mingw-specific fix.

## Proposal

- Collect the pros/cons of why the mingw build is needed.
- If there are no strong "Pro" items discovered as part of this discussion, _remove_ the Windows mingw build from Nvim CI.
    - If there is a strong need to have the compile-checks of mingw, consider only _building_ mingw Nvim in CI, don't run tests.

cc @jamessan

### PR DESCRIPTION
**Pull request by dundargoc**
The mingw CI job takes a long time to finish and the benefit it provides
is relatively low. A more thorough rationale and discussion can be found
in the provided link.

Closes https://github.com/neovim/neovim/issues/18551

### CODE REVIEW

**Review by justinmk — APPROVED** [review]
is there build.ps1 or other script code that can be simplified ? We probably can't continue to support the msys stuff in build.ps1 if we aren't checking it in CI.

**Comment by dundargoc** [comment]
> is there build.ps1 or other script code that can be simplified ? We probably can't continue to support the msys stuff in build.ps1 if we aren't checking it in CI.

Yes. Give me 10 min.

File: ci/build.ps1 [hunk 1]
```diff
@@ -40,44 +40,7 @@ if (-Not (Test-Path -PathType container $env:DEPS_BUILD_DIR)) {
...
-  & C:\msys64\usr\bin\mkdir -p /var/cache/pacman/pkg
-
-  # Build third-party dependencies
-  C:\msys64\usr\bin\bash -lc "pacman --verbose --noconfirm -Syu" ; exitIfFailed
-  # Update again in case updating pacman changes pacman.conf
-  C:\msys64\usr\bin\bash -lc "pacman --verbose --noconfirm -Syu" ; exitIfFailed
-  C:\msys64\usr\bin\bash -lc "pacman --verbose --noconfirm --needed -S $mingwPackages" ; exitIfFailed
-}
-elseif ($compiler -eq 'MSVC') {
+if ($compiler -eq 'MSVC') {
```
Comment (justinmk) [inline ci/build.ps1:43]: maybe out of scope for this PR, but should we throw an error if `$compiler` is *not* MSVC? what else would it be?

File: CMakeLists.txt [hunk 2]
```diff
@@ -108,14 +108,7 @@ endif()
 
 option(ENABLE_LIBINTL "enable libintl" ON)
 option(ENABLE_LIBICONV "enable libiconv" ON)
-if (MINGW)
-  # Disable LTO by default as it may not compile
-  # See https://github.com/Alexpux/MINGW-packages/issues/3516
-  # and https://github.com/neovim/neovim/pull/8654#issuecomment-402316672
-  option(ENABLE_LTO "enable link time optimization" OFF)
-else()
```
Comment (justinmk) [inline CMakeLists.txt:116]: Sorry for the runaround, but we might want to wait on removing mingw stuff from CMakeLists. That's different that build.ps1, which is merely a "quickstart" script. If we remove mingw things from CMakeLists, that makes it very hard for anyone to build with mingw, if they wish.

I'm open to objections though.

File: ci/build.ps1 (hunk 1 above)
Comment (dundargoc) [reply ci/build.ps1:43]: I dunno, I'm very unfamiliar with windows ecosystem :).

I'll just remove the check for the compilers then.

File: third-party/cmake/BuildLuajit.cmake [hunk 3]
```diff
@@ -76,57 +76,6 @@ if(UNIX)
     CC=${DEPS_C_COMPILER} PREFIX=${DEPS_INSTALL_DIR}
     ${DEPLOYMENT_TARGET})
 
-elseif(MINGW AND CMAKE_CROSSCOMPILING)
```
Comment (justinmk) [inline third-party/cmake/BuildLuajit.cmake:79]: This we can definitely get rid of. Not sure if we need crosscompiling stuff at all.

File: CMakeLists.txt (hunk 2 above)
Comment (dundargoc) [reply CMakeLists.txt:116]: Yeah, I was about to ask you about this as well. Makes sense,

**Review by justinmk — APPROVED** [review]
Well, let's go ahead with this as-is and see if anyone cares about the mingw support. We can surgically restore some stuff later if needed.

File: ci/build.ps1 (hunk 1 above)
Comment (justinmk) [reply ci/build.ps1:43]: yeah build.ps1 is just a quickstart script for our own use (for CI, but also potentially for local development).

**Review by justinmk — APPROVED** [review]

**Comment by justinmk** [comment]
Does CI still use msys for running oldtests? https://github.com/neovim/neovim/blob/e501e4ed4ba86da12a9b4747fdc326b8366a38be/ci/build.ps1#L121-L125

**Comment by dundargoc** [comment]
Honestly, I wasn't sure so I just left is as I didn't want to break something acidentally.

**Pull request merged**